packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1417,6 +1417,57 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return child;
   }
 
+  /// The delegated transition provided to the previous route.
+  DelegatedTransitionBuilder? get delegatedTransition => null;
+
+  /// The delegated transition received from an incoming route.
+  DelegatedTransitionBuilder? receivedTransition;
+
+  /// Transition that matches with the next route. Will be null if there is no
+  /// next route or a next route is not provided. If a non-null value exists,
+  /// then it will be used in place of [buildTransitions].
+  DelegatedTransitionBuilder? get nextRouteTransition {
```

Comment:
So this is related to the approach we've been discussing of putting all 3 transition widgets into the widget tree at the same time, right? Can you explain it to me again? I think we need to clarify in the docs about this and buildFlexTransitions. Maybe also think about the naming and how it relates to delegatedTransition.

Reply:
Does this need to be a separate public getter from receivedTransition?